rflx/expression.py

```diff
@@ -62,8 +72,7 @@ def __neg__(self) -> 'Expr':
     def precedence(self) -> Precedence:
         raise NotImplementedError
 
-    @property
-    def variables(self) -> List['Variable']:
+    def variables(self, proof: bool = False) -> List['Variable']:  # pylint: disable=unused-argument,no-self-use
```

Comment:
Because `variables` is not a static method. It gets overridden in classes derived from `Expr` (e.g. BinExpr) where `self` is required. Not sure how to resolve the pylint issue without disabling warnings in such cases.

Reply:
As there is no good alternative, let's stick to the current solution.